src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java

```diff
@@ -83,17 +83,16 @@ public void writeRecord(SinkRecord record) {
             this.currentOffset = record.kafkaOffset();
         } else {
             try {
-                fileWriter.write(value);
-
                 this.currentOffset = record.kafkaOffset();
+                fileWriter.write(value);
```

Comment:
why? shouldn't we update the offset only after a successful write?

Reply:
isn't it used just by us ? to track with file names when sending file to kusto?

Reply:
if so - fileWriter.write  will close the file including the current file but it won't reflect in the file name